base/cvd/cuttlefish/host/commands/cvd/fetch/fetch_tracer.h

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright (C) 2024 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace cuttlefish {
+
+// FetchTracer allows tracking the performance of fetch operations.
+// For each independing fetch, like fetching the host packages, a new trace
+// should be created. Each trace is then split in phases, each of which tracks
+// duration and, optionally, download size. The FetchTracer class is thread
+// safe, the FetchTracer::Trace is not and each trace should only be used from a
+// single thread.
```

Comment:
nit: it would be useful to add a coding example of how the API is meant to be used in the documentation.

Reply:
This is only going to be used from fetch_cvd.cc and there are already examples there.